Source/EPUBCore/FREpubParser.swift

```diff
@@ -21,15 +21,22 @@ class FREpubParser: NSObject, SSZipArchiveDelegate {
     Returns a FRBook.
     */
     func readEpub(epubPath withEpubPath: String) -> FRBook {
+        
         epubPathToRemove = withEpubPath
         
         // Unzip   
         let bookName = (withEpubPath as NSString).lastPathComponent
-        bookBasePath = (kApplicationDocumentsDirectory as NSString).stringByAppendingPathComponent(bookName)
-        SSZipArchive.unzipFileAtPath(withEpubPath, toDestination: bookBasePath, delegate: self)
+        bookBasePath = (kApplicationDocumentsDirectory as NSString).stringByAppendingPathComponent(bookName+".dir")
         
-        // Skip from backup this folder
-        addSkipBackupAttributeToItemAtURL(NSURL(fileURLWithPath: bookBasePath, isDirectory: true))
+        // unzip the epub if we have not already done so
+        var isDir : ObjCBool = true
+        if NSFileManager.defaultManager().fileExistsAtPath(bookBasePath, isDirectory: &isDir) == false {
```

Comment:
Hmmm. With that logic, how come when trying to load the same epub again (from the menu) it crashes because the epub is no longer there? Shouldn't it detect the unzipped directory and use that instead of crash on non-existent epub file?